pgxn/neon/neon.c

```diff
@@ -522,29 +539,44 @@ RestoreRunningXactsFromClog(CheckPoint *checkpoint, TransactionId **xids, int *n
 			elog(LOG, "too many running xacts to restore from the CLOG; oldestXid=%u oldestActiveXid=%u nextXid %u",
 				 checkpoint->oldestXid, checkpoint->oldestActiveXid,
 				 XidFromFullTransactionId(checkpoint->nextXid));
-			goto fail;
+
+			overflow = true;
+			switch (running_xacts_overflow_policy)
+			{
+				case OP_WAIT:
+					goto fail;
+				case OP_IGNORE:
+					goto success;
+				case OP_SKIP:
+					n_restored_xids = 0;
+					goto success;
+			}
 		}
 
 		restored_xids[n_restored_xids++] = xid;
 
 	skip:
 		TransactionIdAdvance(xid);
-		continue;
 	}
 
 	/* sanity check */
 	if (next_prepared_idx != n_prepared_xids)
 	{
 		elog(LOG, "prepared transaction ID %u was not visited in the CLOG scan, cannot restore running-xacts from CLOG",
 			 prepared_xids[next_prepared_idx]);
-		Assert(false);
-		goto fail;
+		if (!overflow)
+		{
+			Assert(false);
+			goto fail;
+		}
```

Comment:
If we now jump to success, do we still need the `overflow` flag and this code? Looks like it's unreachable in case of overflow

Reply:
Thank you, it is not needed any more.